app/controllers/meetings/publish/uittreksels.js

```diff
@@ -0,0 +1,73 @@
+import Controller from '@ember/controller';
+import {inject as service} from "@ember/service";
+import {task} from "ember-concurrency-decorators";
+import {tracked} from "@glimmer/tracking";
+
+export default class MeetingsPublishUittrekselsController extends Controller {
+  @service ajax;
+
+  @tracked
+  uittreksels = [];
+
+  constructor() {
+    super(...arguments);
+  }
+
+  initialize() {
+    this.initializeUittreksels.perform();
+  }
+
+
+  @task
+  * initializeUittreksels() {
+    this.uittreksels = [];
+    const prePublish = yield this.createPrePublishedResource.perform();
+    for(const uittreksel of prePublish) {
+      const existingUittreksels = yield this.store.query('versioned-behandeling',{
+        'filter[behandeling][:id:]': uittreksel.data.attributes.behandeling,
+        include: 'signed-resources,published-resource'
+      });
+      if(existingUittreksels.length) {
+        this.uittreksels.push(existingUittreksels.firstObject);
+      } else {
+        const behandeling = yield this.store.findRecord('behandeling-van-agendapunt', uittreksel.data.attributes.behandeling);
+
+        const rslt = yield this.store.createRecord("versioned-behandeling", {
+          zitting: this.model,
+          content: uittreksel.data.attributes.content,
+          behandeling,
+        });
+        this.uittreksels.push(rslt);
+      }
+    }
+  }
+
+
+  @task
+  *createPrePublishedResource() {
```

Comment:
shouldn't this also get the behandeling as an argument?

Reply:
No, because this one gets the prepublished resource from the service, doesn't need the old behandeling if there is one

Reply:
I keep the naming consistent with the other routes, but maybe this one should be called getPrepublishedBehandelings
As it gets all of them not just one

Reply:
As discussed I think this might not work for larger agenda's (50+ agendapoints). perhaps better to work with a request per agendapoint and only do the request if that one was selected